### PR TITLE
[DOCS] Updated AudioStream docs and added AudioServer docs

### PR DESCRIPTION
**Pull request by Grosskopf**
Changed documentation according to missing Methods and Constants for AudioStreams, added AudioServer and AudioBusLayout documentation

### CODE REVIEW

File: doc/classes/AudioBusLayout.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AudioBusLayout" inherits="Resource" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Stores the information about the audiobusses.
```
Comment (cbscribe) [inline doc/classes/AudioBusLayout.xml]: "Stores information" sounds better.

File: doc/classes/AudioBusLayout.xml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 	</brief_description>
 	<description>
+		Stores position, muting, solo, bypass, effects, effectposition, volume and the way the busses input into each other. [codeblock]See [AudioServer] for usage.[/codeblock]
```
Comment (cbscribe) [inline doc/classes/AudioBusLayout.xml]: Should be "effect position" and you need a comma after volume.  

Also, "the way the busses input into each other" sounds awkward. How about "and the connections between busses."

Comment (Noshyaar) [inline doc/classes/AudioBusLayout.xml]: codeblock?

File: doc/classes/AudioServer.xml [hunk 3]
```diff
@@ -29,18 +30,21 @@
...
 			</return>
 			<description>
+				Generates an [AudioBusLayout] using the busses and effects avaliable.
 			</description>
 		</method>
 		<method name="get_bus_count" qualifiers="const">
 			<return type="int">
 			</return>
 			<description>
+				Returns the amount of busses avaliable.
```
Comment (Noshyaar) [inline doc/classes/AudioServer.xml]: available

File: doc/classes/AudioServer.xml [hunk 4]
```diff
@@ -29,18 +30,21 @@
 			<argument index="2" name="at_position" type="int" default="-1">
 			</argument>
 			<description>
+				Adds an [AudioEffect] effect to the bus at [code]bus_idx[/code] at effectposition [code]at_position[/code].
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: Repeating the words isn't necessary: "Adds an [AudioEffect] to the bus" and "at position [code]at_position[/code]."

File: doc/classes/AudioServer.xml [hunk 5]
```diff
@@ -29,18 +30,21 @@
...
 			</description>
 		</method>
 		<method name="generate_bus_layout" qualifiers="const">
 			<return type="AudioBusLayout">
 			</return>
 			<description>
+				Generates an [AudioBusLayout] using the busses and effects avaliable.
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: "available" is misspelled

"using the available busses and effects"

File: doc/classes/AudioServer.xml [hunk 6]
```diff
@@ -111,18 +122,21 @@
...
 			</return>
 			<description>
+				Returns the samplerate at the output of the audioserver.
 			</description>
 		</method>
 		<method name="get_speaker_mode" qualifiers="const">
 			<return type="int" enum="AudioServer.SpeakerMode">
 			</return>
 			<description>
+				Returns, what kind of speaker setup is detected.
```
Comment (Noshyaar) [inline doc/classes/AudioServer.xml]: without comma I guess

File: doc/classes/AudioServer.xml (hunk 3 above)
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: Returns the number of available busses.

File: doc/classes/AudioServer.xml [hunk 7]
```diff
@@ -259,6 +287,7 @@
 			<argument index="1" name="send" type="String">
 			</argument>
 			<description>
+				Sets the bus towards which the bus at index [code]bus_idx[/code] sends it's output to the bus named [code]send[/code].
```
Comment (Noshyaar) [inline doc/classes/AudioServer.xml]: its

File: doc/classes/AudioServer.xml [hunk 8]
```diff
@@ -59,6 +64,7 @@
 			<argument index="0" name="bus_idx" type="int">
 			</argument>
 			<description>
+				Returns the amount of effects on the bus at [code]bus_idx[/code].
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: amount -> number

File: doc/classes/AudioServer.xml [hunk 9]
```diff
@@ -85,6 +93,7 @@
 			<argument index="1" name="channel" type="int">
 			</argument>
 			<description>
+				Returns the peak volume of the left speaker at bus index [code]bus_idx[/code] channel index [code]channel[/code].
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: add an "and" before "channel index

File: doc/classes/AudioServer.xml [hunk 10]
```diff
@@ -95,6 +104,7 @@
 			<argument index="1" name="channel" type="int">
 			</argument>
 			<description>
+				Returns the peak volume of the right speaker at bus index [code]bus_idx[/code] channel index [code]channel[/code].
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: add an "and" before "channel index"

File: doc/classes/AudioServer.xml [hunk 11]
```diff
@@ -111,18 +122,21 @@
...
 			</argument>
 			<description>
+				Returns the volume of the bus at index [code]bus_idx[/code] in dB.
 			</description>
 		</method>
 		<method name="get_mix_rate" qualifiers="const">
 			<return type="float">
 			</return>
 			<description>
+				Returns the samplerate at the output of the audioserver.
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: "sample rate"

File: doc/classes/AudioServer.xml (hunk 6 above)
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: Returns the speaker configuration.

File: doc/classes/AudioServer.xml [hunk 12]
```diff
@@ -291,27 +322,33 @@
...
 			</return>
 			<description>
+				Unlocks the audiodriver's main loop. After locking it always unlock it.
 			</description>
 		</method>
 	</methods>
 	<signals>
 		<signal name="bus_layout_changed">
 			<description>
+				Emitted, when the [AudioBusLayout] changed.
```
Comment (Noshyaar) [inline doc/classes/AudioServer.xml]: without comma?

File: doc/classes/AudioServer.xml [hunk 13]
```diff
@@ -209,6 +232,7 @@
 			<argument index="0" name="amount" type="int">
 			</argument>
 			<description>
+				Adds and removes busses to make the amount of busses match [code]amount[/code].
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: amount -> number

File: doc/classes/AudioStreamPlayback.xml [hunk 14]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AudioStreamPlayback" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Meta class for playing back audio.
 	</brief_description>
 	<description>
+		Can play, loop, pause and and scroll through Audio. [codeblock]See [AudioStream] and [AudioStreamOGGVorbis] for usage.[/codeblock]
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayback.xml]: Can play, loop, pause, and scroll

File: doc/classes/AudioServer.xml (hunk 7 above)
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: Connects the output of the bus at [code]bus_idx[/code] to the bus named [code]send[/send].

File: doc/classes/AudioServer.xml (hunk 12 above)
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: no comma
changed -> changes

File: doc/classes/AudioServer.xml [hunk 15]
```diff
@@ -291,27 +322,33 @@
...
 	<signals>
 		<signal name="bus_layout_changed">
 			<description>
+				Emitted, when the [AudioBusLayout] changed.
 			</description>
 		</signal>
 	</signals>
 	<constants>
 		<constant name="SPEAKER_MODE_STEREO" value="0">
+			There are two or less speakers detected.
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: less -> fewer

Two or fewer speakers are detected.

File: doc/classes/AudioServer.xml [hunk 16]
```diff
@@ -291,27 +322,33 @@
...
 		</constant>
 		<constant name="SPEAKER_SURROUND_51" value="2">
+			There is a 5.1 surround setup detected.
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: A 5.1 channel surround setup is detected.

File: doc/classes/AudioServer.xml [hunk 17]
```diff
@@ -291,27 +322,33 @@
...
 		</constant>
 		<constant name="SPEAKER_SURROUND_71" value="3">
+			There is a 7.1 surround setup detected.
```
Comment (cbscribe) [inline doc/classes/AudioServer.xml]: A 7.1 channel surround setup is detected.

File: doc/classes/AudioStreamPlayer.xml [hunk 18]
```diff
@@ -27,6 +28,7 @@
 			<return type="float">
 			</return>
 			<description>
+				Returns the position in the [AudioStream] it is playing at in seconds.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer.xml]: Returns the current position in the [AudioStream].

File: doc/classes/AudioStreamPlayer.xml [hunk 19]
```diff
@@ -127,6 +129,7 @@
 			Bus on which this audio is playing.
 		</member>
 		<member name="mix_target" type="int" setter="set_mix_target" getter="get_mix_target" enum="AudioStreamPlayer.MixTarget">
+			In case you have an audio setup with more than stereo speakers, this defines, to which speaker-channels the [member stream] is played.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer.xml]: Too many commas.

If the audio configuration has more than two speakers, this sets the target channels. See [code]MIX_TARGET_*[/code] constants.

File: doc/classes/AudioStreamPlayer.xml [hunk 20]
```diff
@@ -127,6 +129,7 @@
...
 		</member>
 		<member name="playing" type="bool" setter="_set_playing" getter="is_playing">
 			If [code]true[/code], audio is playing.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer.xml]: remove comma

File: doc/classes/AudioStreamPlayer.xml [hunk 21]
```diff
@@ -147,10 +150,13 @@
 	</signals>
 	<constants>
 		<constant name="MIX_TARGET_STEREO" value="0">
+			The audio will be played only on the first channel.
 		</constant>
 		<constant name="MIX_TARGET_SURROUND" value="1">
+			The audio will be played on all surround channels.
 		</constant>
 		<constant name="MIX_TARGET_CENTER" value="2">
+			The audio will be played on the second channel which should be the center.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer.xml]: "...second channel, which is usually the center."

File: doc/classes/AudioStreamPlayer2D.xml [hunk 22]
```diff
@@ -40,6 +40,7 @@
 			<return type="float">
 			</return>
 			<description>
+				Returns the position in the [AudioStream] it is playing at in seconds.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer2D.xml]: Returns the current position in the [AudioStream].

File: doc/classes/AudioStreamPlayer3D.xml [hunk 23]
```diff
@@ -82,6 +82,7 @@
 			<return type="float">
 			</return>
 			<description>
+				Returns the position in the [AudioStream] it is playing at in seconds.
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer3D.xml]: Returns the current position in the [AudioStream].

**Review by cbscribe — CHANGES_REQUESTED** [review]

File: doc/classes/AudioServer.xml (hunk 7 above)
Comment (Grosskopf) [reply doc/classes/AudioServer.xml]: thanks, that's so much less awkward

File: doc/classes/AudioStreamPlayer3D.xml [hunk 24]
```diff
@@ -288,7 +289,7 @@
 			Decides if audio should get quieter with distance linearly, quadratically or logarithmically.
 		</member>
 		<member name="autoplay" type="bool" setter="set_autoplay" getter="is_autoplay_enabled">
-			If [code]true[/code], audio plays audio plays when added to scene tree. Default value: [code]false[/code].
+			If [code]true[/code] audio plays audio plays when added to scene tree. Default value: [code]false[/code].
```
Comment (cbscribe) [inline doc/classes/AudioStreamPlayer3D.xml]: Too many audios?  "audio plays when added"

**Review by cbscribe — APPROVED** [review]

File: doc/classes/AudioStreamPlayer3D.xml (hunk 24 above)
Comment (Grosskopf) [reply doc/classes/AudioStreamPlayer3D.xml]: oops :D

**Comment by Grosskopf** [comment]
thanks a lot :) changed the last one aaand ready to merge :)

**Pull request merged**